### PR TITLE
doc transaction types and object fields for EIPs 2930 and 1559

### LINKED ISSUE #737: Add transaction types page
## Describe the bug

Need content on the different transaction types: legacy, access list, and 1559 transactions.

see https://github.com/hyperledger/besu/pull/1645, https://github.com/hyperledger/besu/pull/1712, and https://github.com/hyperledger/besu/pull/1825

- [ x] Missing content
- [ ] Outdated content
- [ ] Wrong content
- [ ] Confusing or misleading content
- [ ] Other

## The broken page

[Paste the doc site page link here.]

## Change suggestion

 
Add transaction types page to the concepts category. Update API objects. Link access list and 1559 reference content back to this page.

### PR DESCRIPTION
**Pull request by alexandratran**
## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [x] you fixed all the issues raised by the tests, if any.
- [x] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

Update API Objects documentation to include new fields for typed transactions (EIPs 2718, 2930, and 1559).

Add Transaction Types concept page.

## Issue fixed

fixes #737 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing

see https://hyperledger-besu--738.org.readthedocs.build/en/738/Concepts/Transactions/Transaction-Types/
and https://hyperledger-besu--738.org.readthedocs.build/en/738/Reference/API-Objects/

### CODE REVIEW

File: docs/Concepts/Transactions/Transaction-Types.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- [Transaction call object](../../Reference/API-Objects.md#transaction-call-object)
+- [Transaction receipt object](../../Reference/API-Objects.md#transaction-receipt-object)
+
+## `FRONTIER` transactions
+
+Transactions with type `FRONTIER` are *legacy transactions* that use the transaction format existing before typed
+transactions were introduced in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718).
+They contain the parameters `chainId`, `nonce`, `gasPrice`, `gasLimit`, `to`, `value`, `data`, `v`, `r`, and `s`.
+Legacy transactions are not compatible with [access lists](#access_list-transactions) or
+[EIP-1559 fee market changes](#eip1559-transactions).
```
Comment (garyschulte) [inline docs/Concepts/Transactions/Transaction-Types.md:23]: perhaps better stated as 'Legacy transactions are not compatible with parameters introduced by access list transactions or 1559 fee market transactions.  

Legacy transactions will continue to work post-1559, so the usage of 'compatibility' might be confusing.

File: docs/Concepts/Transactions/Transaction-Types.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## `ACCESS_LIST` transactions
+
+Transactions with type `ACCESS_LIST` are transactions introduced in
+[EIP-2930](https://eips.ethereum.org/EIPS/eip-2930).
+They contain, along with the [legacy parameters](#frontier-transactions), an `accessList` parameter, which specifies an
+array of addresses and storage keys that the transaction plans to access (an *access list*).
+`ACCESS_LIST` transactions must specify an access list, and they are not compatible with
+[EIP-1559 fee market changes](#eip1559-transactions).
```
Comment (garyschulte) [inline docs/Concepts/Transactions/Transaction-Types.md:32]: similar thoughts about the use of 'compatibility' here.  Access list txns will continue to work, they just won't benefit from 1559 fee market changes.

File: docs/Concepts/Transactions/Transaction-Types.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Transactions with type `EIP1559` are transactions introduced in
+[EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md).
+EIP-1559 addresses the network congestion and overpricing of transaction fees caused by the historical fee market,
+in which users send transactions specifying a gas price bid using the `gasPrice` parameter, and miners choose
+transactions with the highest bids.
+
+`EIP1559` transactions don't specify `gasPrice`, and instead use an in-protocol, dynamically changing *base fee* per gas.
+At each block, the base fee per gas is adjusted to address network congestion as measured by a gas target.
+
+`EIP1559` transactions contain, along with the [legacy parameters](#frontier-transactions) and
```
Comment (garyschulte) [inline docs/Concepts/Transactions/Transaction-Types.md:45]: legacy parameters other than gasPrice.  You do address this elsewhere though.

File: docs/Reference/API-Objects.md [hunk 4]
```diff
@@ -93,17 +93,21 @@ Returned by [`txpool_besuPendingTransactions`](API-Methods.md#txpool_besupending
...
-| **input**            | Data                | Data sent with the transaction to create or invoke a contract.      |
-| **nonce**            | Quantity            | Number of transactions made by the sender before this one.          |
-| **to**               | Data, 20&nbsp;bytes | Address of the receiver. `null` if a contract creation transaction. |
-| **value**            | Quantity            | Value transferred, in wei.                                          |
-| **v**                | Quantity            | ECDSA Recovery ID.                                                  |
-| **r**                | Data, 32&nbsp;bytes | ECDSA signature r.                                                  |
-| **s**                | Data, 32&nbsp;bytes | ECDSA signature s.                                                  |
+| **accessList**           | Array               | List of addresses and storage keys the transaction plans to access. Used in [`ACCESS_LIST` transactions](../Concepts/Transactions/Transaction-Types.md#access_list-transactions) and may be used in [`EIP1559` transactions](../Concepts/Transactions/Transaction-Types.md#eip1559-transactions). |
```
Comment (garyschulte) [inline docs/Reference/API-Objects.md:96]: do we have a way to mark these as nullable or optional?  same q for gasPrice, maxFeePerGas, maxPriorityFeePerGas

**Review by garyschulte — CHANGES_REQUESTED** [review]
Looks good, minor feedback about 'compatibility'

File: CI/vale/vale_styles/Vocab/Besu/accept.txt [hunk 5]
```diff
@@ -57,6 +57,7 @@ Helmfile(s)?
 Homebrew
 [hH]ostname(s)?
 Hyperledger
+[iI]ncentiviz(e|es|ed|ing)
```
Comment (NicolasMassart) [inline CI/vale/vale_styles/Vocab/Besu/accept.txt:60]: Regex should not contain capture groups and be
```
[iI]ncentiviz(?:e|es|ed|ing)
```

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Review by garyschulte — APPROVED** [review]
LGTM, thanks

**Pull request merged**